tests/components/music_assistant/test_media_player.py

```diff
@@ -0,0 +1,176 @@
+        {
+            "entity_id": entity_id,
+            "volume_level": 0.5,
+        },
+        blocking=True,
+    )
+    assert music_assistant_client.send_command.call_count == 1
+    assert music_assistant_client.send_command.call_args == call(
+        "players/cmd/volume_set", player_id=mass_player_id, volume_level=50
+    )
+    music_assistant_client.send_command.reset_mock()
+
+    # test volume mute action
+    await hass.services.async_call(
+        "media_player",
+        "volume_mute",
+        {
+            "entity_id": entity_id,
+            "is_volume_muted": True,
+        },
+        blocking=True,
+    )
+    assert music_assistant_client.send_command.call_count == 1
+    assert music_assistant_client.send_command.call_args == call(
+        "players/cmd/volume_mute", player_id=mass_player_id, muted=True
+    )
+    music_assistant_client.send_command.reset_mock()
+
+    # test turn_on /turn_off action
+    for action, pwr in (
+        ("turn_on", True),
+        ("turn_off", False),
+    ):
+        await hass.services.async_call(
+            "media_player",
+            action,
+            {
+                "entity_id": entity_id,
+            },
+            blocking=True,
+        )
+        assert music_assistant_client.send_command.call_count == 1
+        assert music_assistant_client.send_command.call_args == call(
+            "players/cmd/power", player_id=mass_player_id, powered=pwr
+        )
+        music_assistant_client.send_command.reset_mock()
+
+    # test shuffle action
+    await hass.services.async_call(
+        "media_player",
+        "shuffle_set",
+        {
+            "entity_id": entity_id,
+            "shuffle": True,
+        },
+        blocking=True,
+    )
+    assert music_assistant_client.send_command.call_count == 1
+    assert music_assistant_client.send_command.call_args == call(
+        "player_queues/shuffle", queue_id=mass_player_id, shuffle_enabled=True
+    )
+    music_assistant_client.send_command.reset_mock()
+
+    # test repeat action
+    await hass.services.async_call(
+        "media_player",
+        "repeat_set",
+        {
+            "entity_id": entity_id,
+            "repeat": "one",
+        },
+        blocking=True,
+    )
+    assert music_assistant_client.send_command.call_count == 1
+    assert music_assistant_client.send_command.call_args == call(
+        "player_queues/repeat", queue_id=mass_player_id, repeat_mode="one"
+    )
+    music_assistant_client.send_command.reset_mock()
+
+    # test clear playlist action
+    await hass.services.async_call(
+        "media_player",
+        "clear_playlist",
+        {
+            "entity_id": entity_id,
+        },
+        blocking=True,
+    )
+    assert music_assistant_client.send_command.call_count == 1
+    assert music_assistant_client.send_command.call_args == call(
+        "player_queues/clear", queue_id=mass_player_id
+    )
+    music_assistant_client.send_command.reset_mock()
```

Comment:
Why don't we split this into separate tests?

Reply:
You mean per action a separate test ?
So a test for the repeat action, clear playlist etc.

No particular reason other than performance maybe. I don't mind splitting it up if that makes you happy

Reply:
I think I want them split. Not because I have a major reason, but more that I think it makes sense to have, because otherwise we have very long and large tests